### DISCUSSION TITLE
authentication & authorization in mockcoon

### DISCUSSION THREAD

**Discussion by jacksonp2008** [mockoon/mockoon#390] [Help] [answered]
Do you support authentication in mockcoon?  (I suppose we can just look at the authorization header for basic)

Your notes mention you support swagger/openapi which do support a variety https://swagger.io/docs/specification/authentication/

thanks

**Comment by 255kb** [comment, marked as answer]
For the moment, Mockoon doesn't implement any kind of logic for authentication. You can send whatever you want when calling the mock endpoints. 
You may still react to the presence of an Authorization header with the [rules](https://mockoon.com/docs/latest/multiple-responses/) or the [templating system](https://mockoon.com/docs/latest/templating/), and serve different responses (200, 500, etc.) or contents .

If you have any ideas of what such feature could look like, they are welcome 😃

**Reply by jacksonp2008** [reply]
Thanks, this is a very useful tool.

**Reply by 255kb** [reply]
You're welcome 👍
